### PR TITLE
CONTRIB: add info on how to find PR preview html link

### PR DESCRIPTION
**Pull request by rgerhards**

### CODE REVIEW

**Comment by deoren** [comment]
@rgerhards Is it possible to generate the directory based off of the git commit id? I'm wondering what the chances of collisions are if two PRs are submitted close together. Probably not too likely, but I figured it was worth asking.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -21,6 +21,23 @@ of the module documentation.
 
 For links, the same policies as laid down below applies.
 
+# Reviewing your PRs
+
+After submitting your pull request, please check for CI errors and handle
+them. **PRs with CI failures cannot be matched.**
```
Comment (deoren) [inline CONTRIBUTING.md]: Did you mean to say "merged" instead of matched?

**Comment by rgerhards** [comment]
Git commit is more work, and collision is impossible as the testbench runs
serially.

Rainer

Sent from phone, thus brief.

**Comment by deoren** [comment]
> @rgerhards: Git commit is more work, and collision is impossible as the testbench runs
serially.

ACK.

**Comment by rgerhards** [comment]
Sent from phone, thus brief.

Am 13.02.2018 18:17 schrieb "Deoren Moor" <notifications@github.com>:

*@deoren* commented on this pull request.
------------------------------

In CONTRIBUTING.md
<https://github.com/rsyslog/rsyslog-doc/pull/567#discussion_r167936781>:

> @@ -21,6 +21,23 @@ of the module documentation.

 For links, the same policies as laid down below applies.

+# Reviewing your PRs
+
+After submitting your pull request, please check for CI errors and handle
+them. **PRs with CI failures cannot be matched.**

Did you mean to say "merged" instead of matched?


Lol, indeed.

**Comment by rgerhards** [comment]
> @rgerhards: Git commit is more work, and collision is impossible as the testbench runs
> serially.
> 
> ACK.

actually, I was wrong ;-)
```
git log -1 --pretty=format:"%H"
```
gives the head commit, albeit I am not sure it is the same thing that'll show up in CI. Note that I use %H instead of %h (short hash) as in theory the latter can have duplicates. I admit I am still not sure if I feel better about the hash than the date - I admit I can assign more meaning to the date than to the hash If I just see it...

**Comment by rgerhards** [comment]
> actually, I was wrong ;-)
> 
> git log -1 --pretty=format:"%H"
> gives the head commit, albeit I am not sure it is the same thing that'll show up in CI. Note that I use %H instead of %h (short hash) as in theory the latter can have duplicates. I admit I am still not sure if I feel better about the hash than the date - I admit I can assign more meaning to the date than to the hash If I just see it...

yeah, it's the same hash in CI (I speculated it might generate a temporary commit). I have right now changed the buildbot conf to use it, so you can have a look (in this PR right now). Maybe we should keep it and see if it makes any difference at all -- just so to prove that I tend to keep concerns that no really apply ;-)

**Comment by deoren** [comment]
> @rgerhards: Note that I use %H instead of %h (short hash) as in theory the latter can have duplicates. I admit I am still not sure if I feel better about the hash than the date - I admit I can assign more meaning to the date than to the hash If I just see it...

What about date + short hash?

The idea being that contributors could visit http://ubuntu16.rsyslog.com/doc/ directly and pick their build from the list to preview the docs. The short hash could be used from the PR to help locate the build on that auto-indexed listing of content.

buildbot-docs-autoindexed-listing-2018-02-13

I forgot to mention it in my earlier reply, but this is a really useful addition to the dev workflow.

**Comment by rgerhards** [comment]
Sent from phone, thus brief.

Am 13.02.2018 19:41 schrieb "Deoren Moor" <notifications@github.com>:

@rgerhards <https://github.com/rgerhards>: Note that I use %H instead of %h
(short hash) as in theory the latter can have duplicates. I admit I am
still not sure if I feel better about the hash than the date - I admit I
can assign more meaning to the date than to the hash If I just see it...

What about date + short hash?

The idea being that contributors could visit http://ubuntu16.rsyslog.com/
doc/ directly and pick their build from the list to preview the docs. The
short hash could be used from the PR to help locate the build on that
auto-indexed listing of content.


That URL is not stable, it can change when we need to shuffle things inside
the CI infrastructure. Seldom, but definitely not something I want to be
documented.

Do you think a contributor will wait for the CI system to build? At least I
am far too impatient, I prefer the 1-second local turnover ;-) is that just
me? I so far have seen it primerely as an aid for merging.

**Comment by deoren** [comment]
> @rgerhards: That URL is not stable, it can change when we need to shuffle things inside
the CI infrastructure. Seldom, but definitely not something I want to be
documented.

Probably not worth a CNAME record then to create a stable URL if the plan is to have users come through the CI results list.

> @rgerhards: Do you think a contributor will wait for the CI system to build? 

I imagine there are cases where I would, but maybe not as a rule.

> @rgerhards: At least I am far too impatient, I prefer the 1-second local turnover ;-) is that just
me? I so far have seen it primerely as an aid for merging.

I do a lot of local quick builds too just to check for syntax errors if nothing else. I guess if I'm really honest with myself, I will probably mainly rely on the buildbot builds when proofing work submitted by others, though I think I found one case where the build results were not as expected. I plan to double-check that tomorrow to make sure I wasn't looking at the results wrong.

I had hoped to work on the PRs in the queue tonight, but other things came up. I'll circle back to them tomorrow unless you opt to process them first. I'm happy to do so, I've just run out of time tonight.

**Comment by rgerhards** [comment]
@deoren I have changed the URL as suggest, but please do not add the URL directly to the doc. The way to get to it is to use the "preview_url" link provided by CI.

I'll that I can look at at least some of the PRs.

**Pull request merged**